{"cells":[{"cell_type":"code","execution_count":1,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":24607,"status":"ok","timestamp":1746507622445,"user":{"displayName":"Kutay Eroğlu","userId":"12425207516295289282"},"user_tz":-180},"id":"lmmxeiyM1Ua5","outputId":"2c632f0a-6d10-405f-eacc-4a8220343186"},"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n"]}],"source":["from google.colab import drive\n","drive.mount(\"/content/drive\")"]},{"cell_type":"code","execution_count":30,"metadata":{"executionInfo":{"elapsed":42,"status":"ok","timestamp":1746514468983,"user":{"displayName":"Kutay Eroğlu","userId":"12425207516295289282"},"user_tz":-180},"id":"NvlHd1scSA6k"},"outputs":[],"source":["# ---- paths --------------------------------------------------------------\n","CLASSIFIER_PATH = \"/content/drive/MyDrive/trained_models/resnet_ffhq_ls.pt\"\n","DEX_CLASSIFIER_PATH = \"/content/drive/MyDrive/trained_models/dex_imdb_wiki.caffemodel.pt\"\n","CSV_PATH        = \"/content/drive/MyDrive/datasets/FFHQaging_LS/FFHQaging.csv\"\n","IMG_DIR         = \"/content/drive/MyDrive/datasets/FFHQaging_LS/images\"\n","REPO_DIR        = \"/content/CUSP\""]},{"cell_type":"code","source":["###########################\n","## Sanity check for data ##\n","###########################\n","\n","# import os, glob, textwrap, subprocess, sys, pathlib\n","\n","# DATA_DIR = \"/content/drive/MyDrive/datasets/FFHQaging_LS/images\"\n","\n","# print(\"DATA_DIR           :\", DATA_DIR)\n","# print(\"Exists on disk?    :\", os.path.isdir(DATA_DIR))\n","# print(\"# *.png files found:\", len(glob.glob(os.path.join(DATA_DIR, \"*.png\"))))\n","# print(\"# *.jpg files found:\", len(glob.glob(os.path.join(DATA_DIR, \"*.jpg\"))))\n","\n","# # peek at a couple of names\n","# for f in sorted(glob.glob(os.path.join(DATA_DIR, \"*\"))[:5]):\n","#     print(\" \", os.path.basename(f))"],"metadata":{"id":"GzEYoyCMWsky","executionInfo":{"status":"ok","timestamp":1746513999076,"user_tz":-180,"elapsed":41,"user":{"displayName":"Kutay Eroğlu","userId":"12425207516295289282"}}},"execution_count":20,"outputs":[]},{"cell_type":"code","execution_count":31,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":114,"status":"ok","timestamp":1746514472804,"user":{"displayName":"Kutay Eroğlu","userId":"12425207516295289282"},"user_tz":-180},"id":"jyle4gVEMfwD","outputId":"cbbb2835-e275-44fd-f2bb-eda9d91bbb25"},"outputs":[{"output_type":"stream","name":"stdout","text":["fatal: destination path '/content/CUSP' already exists and is not an empty directory.\n"]}],"source":["!git clone https://github.com/kutayeroglu/CUSP.git $REPO_DIR\n","%cd /content/CUSP"]},{"cell_type":"code","source":["!cp -f \"$DEX_CLASSIFIER_PATH\" dex_imdb_wiki.caffemodel.pt\n","!ls -l dex_imdb_wiki.caffemodel.pt # Verify that file's in place"],"metadata":{"id":"vvq41uBRXHbI","executionInfo":{"status":"ok","timestamp":1746514549829,"user_tz":-180,"elapsed":5096,"user":{"displayName":"Kutay Eroğlu","userId":"12425207516295289282"}}},"execution_count":36,"outputs":[]},{"cell_type":"code","execution_count":7,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":70365,"status":"ok","timestamp":1746508153101,"user":{"displayName":"Kutay Eroğlu","userId":"12425207516295289282"},"user_tz":-180},"id":"vk968hSrSOdL","outputId":"3be73639-05d3-4332-9e35-ec2d6d40af19"},"outputs":[{"output_type":"stream","name":"stdout","text":["\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m5.6/5.6 MB\u001b[0m \u001b[31m48.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m363.4/363.4 MB\u001b[0m \u001b[31m3.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m13.8/13.8 MB\u001b[0m \u001b[31m113.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m24.6/24.6 MB\u001b[0m \u001b[31m91.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m883.7/883.7 kB\u001b[0m \u001b[31m51.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m664.8/664.8 MB\u001b[0m \u001b[31m1.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m211.5/211.5 MB\u001b[0m \u001b[31m11.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m56.3/56.3 MB\u001b[0m \u001b[31m41.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m127.9/127.9 MB\u001b[0m \u001b[31m19.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m207.5/207.5 MB\u001b[0m \u001b[31m4.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m21.1/21.1 MB\u001b[0m \u001b[31m103.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m781.3/781.3 kB\u001b[0m \u001b[31m52.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","pandas-gbq 0.28.0 requires google-auth-oauthlib>=0.7.0, but you have google-auth-oauthlib 0.4.6 which is incompatible.\n","tensorflow 2.18.0 requires tensorboard<2.19,>=2.18, but you have tensorboard 2.12.0 which is incompatible.\u001b[0m\u001b[31m\n","\u001b[0m"]}],"source":["!pip install -q torch torchvision pillow click tqdm tensorboard==2.12 opencv-python-headless"]},{"cell_type":"code","execution_count":38,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":36154,"status":"ok","timestamp":1746514617490,"user":{"displayName":"Kutay Eroğlu","userId":"12425207516295289282"},"user_tz":-180},"id":"EeYYCVojNHvL","outputId":"7091ff39-51ad-47e3-97a1-a89c48e83c1c"},"outputs":[{"output_type":"stream","name":"stdout","text":["\n","Training options:\n","{\n","  \"gb\": \"Fixed\",\n","  \"msk\": \"Reversed\",\n","  \"num_gpus\": 1,\n","  \"gpu_list\": [\n","    \"0\"\n","  ],\n","  \"image_snapshot_ticks\": 1,\n","  \"network_snapshot_ticks\": 20,\n","  \"random_seed\": 0,\n","  \"training_set_kwargs\": {\n","    \"class_name\": \"training.dataset.ImageCSVDataset\",\n","    \"csv_path\": \"/content/drive/MyDrive/datasets/FFHQaging_LS/FFHQaging.csv\",\n","    \"image_path\": \"/content/drive/MyDrive/datasets/FFHQaging_LS/images\",\n","    \"is_train\": true,\n","    \"cmap_pre_kind\": \"number\",\n","    \"transforms\": \"ffhq_lat\",\n","    \"resolution\": 256,\n","    \"use_labels\": true,\n","    \"max_size\": 60000\n","  },\n","  \"test_set_kwargs\": {\n","    \"class_name\": \"training.dataset.ImageCSVDataset\",\n","    \"csv_path\": \"/content/drive/MyDrive/datasets/FFHQaging_LS/FFHQaging.csv\",\n","    \"image_path\": \"/content/drive/MyDrive/datasets/FFHQaging_LS/images\",\n","    \"is_train\": false,\n","    \"cmap_pre_kind\": \"number\",\n","    \"transforms\": \"ffhq_lat\"\n","  },\n","  \"data_loader_kwargs\": {\n","    \"pin_memory\": true,\n","    \"num_workers\": 3,\n","    \"prefetch_factor\": 2\n","  },\n","  \"G_kwargs\": {\n","    \"class_name\": \"training.networks.Image2Image\",\n","    \"n_downsample\": 5,\n","    \"mapping_kwargs\": {},\n","    \"skip_layers\": [\n","      2\n","    ],\n","    \"skip_kind\": \"linear\",\n","    \"synthesis_kwargs\": {\n","      \"channel_max\": 512,\n","      \"bias\": false,\n","      \"rgb_attention\": false,\n","      \"num_fp16_res\": 4,\n","      \"conv_clamp\": 256\n","    },\n","    \"blur_skip\": true,\n","    \"skip_grad_blur\": \"gb\",\n","    \"learn_mask\": null,\n","    \"attr_dim\": 10,\n","    \"blur_msk\": \"random\",\n","    \"style_enc\": true,\n","    \"bottleneck_class\": false,\n","    \"finetune\": null,\n","    \"style_layers\": 8,\n","    \"style_dim\": 512,\n","    \"attr_layers\": 8,\n","    \"classifier_kwargs\": {\n","      \"class_name\": \"training.loss.DEXAgeClassifier\",\n","      \"vgg_path\": \"dex_imdb_wiki.caffemodel.pt\",\n","      \"own_relu\": true\n","    }\n","  },\n","  \"D_kwargs\": {\n","    \"class_name\": \"training.networks.Discriminator\",\n","    \"block_kwargs\": {},\n","    \"mapping_kwargs\": {},\n","    \"epilogue_kwargs\": {\n","      \"mbstd_num_channels\": 0\n","    },\n","    \"finetune\": null,\n","    \"channel_max\": 512,\n","    \"num_fp16_res\": 4,\n","    \"conv_clamp\": 256\n","  },\n","  \"G_opt_kwargs\": {\n","    \"class_name\": \"torch.optim.Adam\",\n","    \"lr\": 0.0025,\n","    \"betas\": [\n","      0,\n","      0.99\n","    ],\n","    \"eps\": 1e-08\n","  },\n","  \"D_opt_kwargs\": {\n","    \"class_name\": \"torch.optim.Adam\",\n","    \"lr\": 0.0025,\n","    \"betas\": [\n","      0,\n","      0.99\n","    ],\n","    \"eps\": 1e-08\n","  },\n","  \"loss_kwargs\": {\n","    \"class_name\": \"training.loss.I2ILoss\",\n","    \"r1_gamma\": 0.56,\n","    \"loss_weights\": {\n","      \"rec\": 10,\n","      \"cla\": 0.1,\n","      \"adv\": 1,\n","      \"rgb\": 1,\n","      \"act\": 1,\n","      \"msk\": 0.1,\n","      \"msk_smooth\": 1e-05,\n","      \"cam\": 10,\n","      \"fre\": 0.0,\n","      \"cycle\": 10.0\n","    },\n","    \"rgb_reg\": null,\n","    \"act_reg\": \"l2\",\n","    \"mixing_probability\": 0.0,\n","    \"disc_class\": true,\n","    \"classifier_kwargs\": {\n","      \"class_name\": \"training.loss.ResNetClassifier\",\n","      \"model_path\": \"/content/drive/MyDrive/trained_models/resnet_ffhq_ls.pt\",\n","      \"mask_path\": null,\n","      \"outclass\": \"one\"\n","    },\n","    \"random_class_kwargs\": {\n","      \"class_name\": \"training.loss.ChangeAll\"\n","    },\n","    \"classification_loss_kwargs\": {\n","      \"class_name\": \"torch.nn.BCEWithLogitsLoss\",\n","      \"reduction\": \"mean\"\n","    }\n","  },\n","  \"total_kimg\": 25000,\n","  \"batch_size\": 14,\n","  \"batch_gpu\": 14,\n","  \"ema_kimg\": 5,\n","  \"ema_rampup\": null,\n","  \"run_dir\": \"./training-runs-ffhqls/00003-images-256\"\n","}\n","\n","Output directory:   ./training-runs-ffhqls/00003-images-256\n","Training data:      /content/drive/MyDrive/datasets/FFHQaging_LS/images\n","Training duration:  25000 kimg\n","Number of GPUs:     1\n","Number of images:   60000\n","Image resolution:   256\n","Conditional model:  True\n","\n","Creating output directory...\n","Launching processes...\n","Pass-through kwargs {'gb': 'Fixed', 'msk': 'Reversed'}\n","Loading training set...\n","/usr/local/lib/python3.11/dist-packages/torch/utils/data/sampler.py:77: UserWarning: `data_source` argument is not used and will be removed in 2.2.0.You may still have custom implementation that utilizes it.\n","  warnings.warn(\n","\n","Num images:  60000\n","Image shape: [3, 256, 256]\n","Label shape: [10]\n","\n","Constructing networks...\n","/content/CUSP/training/networks.py:1520: UserWarning: Creating a tensor from a list of numpy.ndarrays is extremely slow. Please consider converting the list to a single numpy.ndarray with numpy.array() before converting to a tensor. (Triggered internally at /pytorch/torch/csrc/utils/tensor_new.cpp:254.)\n","  torch.tensor(\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1798, in _jit_compile\n","    _write_ninja_file_and_build_library(\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1888, in _write_ninja_file_and_build_library\n","    verify_ninja_availability()\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1945, in verify_ninja_availability\n","    raise RuntimeError(\"Ninja is required to load C++ extensions\")\n","RuntimeError: Ninja is required to load C++ extensions\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/bias_act.py:50: UserWarning: Failed to build CUDA kernels for bias_act. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/bias_act.py\", line 48, in _init\n","    _plugin = custom_ops.get_plugin('bias_act_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1798, in _jit_compile\n","    _write_ninja_file_and_build_library(\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1888, in _write_ninja_file_and_build_library\n","    verify_ninja_availability()\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1945, in verify_ninja_availability\n","    raise RuntimeError(\"Ninja is required to load C++ extensions\")\n","RuntimeError: Ninja is required to load C++ extensions\n","\n","  warnings.warn('Failed to build CUDA kernels for bias_act. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","\n","Image2Image           Parameters  Buffers  Output shape         Datatype\n","---                   ---         ---      ---                  ---     \n","content_enc.conv0     1792        16       [14, 64, 128, 128]   float32 \n","content_enc.conv1     73856       16       [14, 128, 64, 64]    float32 \n","content_enc.conv2     295168      16       [14, 256, 32, 32]    float32 \n","content_enc.conv3     1180160     16       [14, 512, 16, 16]    float32 \n","content_enc.conv4     2359808     16       [14, 512, 8, 8]      float32 \n","style_enc.conv0       1792        16       [14, 64, 128, 128]   float32 \n","style_enc.conv1       73856       16       [14, 128, 64, 64]    float32 \n","style_enc.conv2       295168      16       [14, 256, 32, 32]    float32 \n","style_enc.conv3       1180160     16       [14, 512, 16, 16]    float32 \n","style_enc.conv4       2359808     16       [14, 512, 8, 8]      float32 \n","style_map.fc0         262656      -        [14, 512]            float32 \n","style_map.fc1         262656      -        [14, 512]            float32 \n","style_map.fc2         262656      -        [14, 512]            float32 \n","style_map.fc3         262656      -        [14, 512]            float32 \n","style_map.fc4         262656      -        [14, 512]            float32 \n","style_map.fc5         262656      -        [14, 512]            float32 \n","style_map.fc6         262656      -        [14, 512]            float32 \n","style_map.fc7         262656      -        [14, 512]            float32 \n","style_map             -           512      [14, 10, 512]        float32 \n","attr_map.fc0          5632        -        [14, 512]            float32 \n","attr_map.fc1          262656      -        [14, 512]            float32 \n","attr_map.fc2          262656      -        [14, 512]            float32 \n","attr_map.fc3          262656      -        [14, 512]            float32 \n","attr_map.fc4          262656      -        [14, 512]            float32 \n","attr_map.fc5          262656      -        [14, 512]            float32 \n","attr_map.fc6          262656      -        [14, 512]            float32 \n","attr_map.fc7          262656      -        [14, 512]            float32 \n","attr_map              -           512      [14, 5, 512]         float32 \n","skip_grad_blur        -           -        [14, 101]            float32 \n","image_dec.b16.conv0   2621441     272      [14, 512, 16, 16]    float32 \n","image_dec.b16.conv1   2621441     272      [14, 512, 16, 16]    float32 \n","image_dec.b16.torgb   264195      -        [14, 3, 16, 16]      float32 \n","image_dec.b16:0       -           16       [14, 512, 16, 16]    float32 \n","image_dec.b16:1       -           -        [14, 512, 16, 16]    float32 \n","image_dec.b32.conv0   2621441     1040     [14, 512, 32, 32]    float16 \n","image_dec.b32.conv1   2621441     1040     [14, 512, 32, 32]    float16 \n","image_dec.b32.torgb   264195      -        [14, 3, 32, 32]      float16 \n","image_dec.b32:0       -           16       [14, 512, 32, 32]    float16 \n","image_dec.b32:1       -           -        [14, 512, 32, 32]    float32 \n","image_dec.b64.conv0   2621441     4112     [14, 512, 64, 64]    float16 \n","image_dec.b64.conv1   2621441     4112     [14, 512, 64, 64]    float16 \n","image_dec.b64.torgb   264195      -        [14, 3, 64, 64]      float16 \n","image_dec.b64:0       -           16       [14, 512, 64, 64]    float16 \n","image_dec.b64:1       -           -        [14, 512, 64, 64]    float32 \n","image_dec.b128.conv0  1802241     16400    [14, 256, 128, 128]  float16 \n","image_dec.b128.conv1  720897      16400    [14, 256, 128, 128]  float16 \n","image_dec.b128.torgb  132099      -        [14, 3, 128, 128]    float16 \n","image_dec.b128:0      -           16       [14, 256, 128, 128]  float16 \n","image_dec.b128:1      -           -        [14, 256, 128, 128]  float32 \n","image_dec.b256.conv0  425985      65552    [14, 128, 256, 256]  float16 \n","image_dec.b256.conv1  212993      65552    [14, 128, 256, 256]  float16 \n","image_dec.b256.torgb  66051       -        [14, 3, 256, 256]    float16 \n","image_dec.b256:0      -           16       [14, 128, 256, 256]  float16 \n","image_dec.b256:1      -           -        [14, 128, 256, 256]  float32 \n","<top-level>           -           451584   [14, 3, 256, 256]    float32 \n","---                   ---         ---      ---                  ---     \n","Total                 31648537    627600   -                    -       \n","\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/upfirdn2d.py:34: UserWarning: Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n","\n","Traceback (most recent call last):\n","  File \"/content/CUSP/torch_utils/ops/upfirdn2d.py\", line 32, in _init\n","    _plugin = custom_ops.get_plugin('upfirdn2d_plugin', sources=sources, extra_cuda_cflags=['--use_fast_math'])\n","              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/custom_ops.py\", line 110, in get_plugin\n","    torch.utils.cpp_extension.load(name=module_name, verbose=verbose_build, sources=sources, **build_kwargs)\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1380, in load\n","    return _jit_compile(\n","           ^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 1823, in _jit_compile\n","    return _import_module_from_library(name, build_directory, is_python_module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/torch/utils/cpp_extension.py\", line 2245, in _import_module_from_library\n","    module = importlib.util.module_from_spec(spec)\n","             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"<frozen importlib._bootstrap>\", line 573, in module_from_spec\n","  File \"<frozen importlib._bootstrap_external>\", line 1233, in create_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","ImportError: /root/.cache/torch_extensions/py311_cu124/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory\n","\n","  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\\n\\n' + traceback.format_exc())\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","/content/CUSP/torch_utils/ops/conv2d_gradfix.py:55: UserWarning: conv2d_gradfix not supported on PyTorch 2.6.0+cu124. Falling back to torch.nn.functional.conv2d().\n","  warnings.warn(f'conv2d_gradfix not supported on PyTorch {torch.__version__}. Falling back to torch.nn.functional.conv2d().')\n","\n","Discriminator  Parameters  Buffers  Output shape         Datatype\n","---            ---         ---      ---                  ---     \n","b256.fromrgb   512         16       [14, 128, 256, 256]  float16 \n","b256.skip      32768       16       [14, 256, 128, 128]  float16 \n","b256.conv0     147584      16       [14, 128, 256, 256]  float16 \n","b256.conv1     295168      16       [14, 256, 128, 128]  float16 \n","b256           -           16       [14, 256, 128, 128]  float16 \n","b128.skip      131072      16       [14, 512, 64, 64]    float16 \n","b128.conv0     590080      16       [14, 256, 128, 128]  float16 \n","b128.conv1     1180160     16       [14, 512, 64, 64]    float16 \n","b128           -           16       [14, 512, 64, 64]    float16 \n","b64.skip       262144      16       [14, 512, 32, 32]    float16 \n","b64.conv0      2359808     16       [14, 512, 64, 64]    float16 \n","b64.conv1      2359808     16       [14, 512, 32, 32]    float16 \n","b64            -           16       [14, 512, 32, 32]    float16 \n","b32.skip       262144      16       [14, 512, 16, 16]    float16 \n","b32.conv0      2359808     16       [14, 512, 32, 32]    float16 \n","b32.conv1      2359808     16       [14, 512, 16, 16]    float16 \n","b32            -           16       [14, 512, 16, 16]    float16 \n","b16.skip       262144      16       [14, 512, 8, 8]      float16 \n","b16.conv0      2359808     16       [14, 512, 16, 16]    float16 \n","b16.conv1      2359808     16       [14, 512, 8, 8]      float16 \n","b16            -           16       [14, 512, 8, 8]      float16 \n","b8.skip        262144      16       [14, 512, 4, 4]      float16 \n","b8.conv0       2359808     16       [14, 512, 8, 8]      float16 \n","b8.conv1       2359808     16       [14, 512, 4, 4]      float16 \n","b8             -           16       [14, 512, 4, 4]      float16 \n","mapping.embed  1024        -        [14, 512]            float32 \n","mapping.fc0    262656      -        [14, 512]            float32 \n","mapping.fc1    262656      -        [14, 512]            float32 \n","mapping.fc2    262656      -        [14, 512]            float32 \n","mapping.fc3    262656      -        [14, 512]            float32 \n","mapping.fc4    262656      -        [14, 512]            float32 \n","mapping.fc5    262656      -        [14, 512]            float32 \n","mapping.fc6    262656      -        [14, 512]            float32 \n","mapping.fc7    262656      -        [14, 512]            float32 \n","b4.conv        2359808     16       [14, 512, 4, 4]      float32 \n","b4.fc          4194816     -        [14, 512]            float32 \n","b4.out         262656      -        [14, 512]            float32 \n","b4             -           -        [14, 1]              float32 \n","---            ---         ---      ---                  ---     \n","Total          31223936    416      -                    -       \n","\n","Distributing across 1 GPUs...\n","Setting up training phases...\n","Traceback (most recent call last):\n","  File \"/content/CUSP/train.py\", line 639, in <module>\n","    main() # pylint: disable=no-value-for-parameter\n","    ^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/click/core.py\", line 1161, in __call__\n","    return self.main(*args, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/click/core.py\", line 1082, in main\n","    rv = self.invoke(ctx)\n","         ^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/click/core.py\", line 1443, in invoke\n","    return ctx.invoke(self.callback, **ctx.params)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/click/core.py\", line 788, in invoke\n","    return __callback(*args, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/click/decorators.py\", line 33, in new_func\n","    return f(get_current_context(), *args, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/train.py\", line 632, in main\n","    subprocess_fn(rank=0, args=args, temp_dir=temp_dir)\n","  File \"/content/CUSP/train.py\", line 492, in subprocess_fn\n","    training_loop.training_loop(rank=rank, gpu_id=gpu_id, **args)\n","  File \"/content/CUSP/training/training_loop.py\", line 213, in training_loop\n","    loss = dnnlib.util.construct_class_by_name(device=device, **ddp_modules, **loss_kwargs) # subclass of training.loss.Loss\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/dnnlib/util.py\", line 289, in construct_class_by_name\n","    return call_func_by_name(*args, func_name=class_name, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/dnnlib/util.py\", line 284, in call_func_by_name\n","    return func_obj(*args, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/training/loss.py\", line 537, in __init__\n","    self.classifier = dnnlib.util.construct_class_by_name(**classifier_kwargs).to(device)\n","                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/dnnlib/util.py\", line 289, in construct_class_by_name\n","    return call_func_by_name(*args, func_name=class_name, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/dnnlib/util.py\", line 284, in call_func_by_name\n","    return func_obj(*args, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/content/CUSP/torch_utils/persistence.py\", line 104, in __init__\n","    super().__init__(*args, **kwargs)\n","  File \"/content/CUSP/training/loss.py\", line 449, in __init__\n","    super().__init__(outclass=outclass)\n","  File \"/content/CUSP/torch_utils/persistence.py\", line 104, in __init__\n","    super().__init__(*args, **kwargs)\n","  File \"/content/CUSP/training/loss.py\", line 341, in __init__\n","    raise NotImplementedError\n","NotImplementedError\n"]}],"source":["!python train.py \\\n","        --csv  {CSV_PATH} \\\n","        --data {IMG_DIR} \\\n","        --classifier_path {CLASSIFIER_PATH} \\\n","        --outdir ./training-runs-ffhqls \\\n","        --gpus 0 --cfg 256 --dataset ffhq_lat \\\n","        --downsamples 5 --bias False --class_w 0.1 \\\n","        --skip_layers 2 --skip_kind linear --act_reg l2 \\\n","        --skip_grad_blur gb --blur_skip true --blur_msk random \\\n","        --mixing_prob 0.0 --disc_class true --fake_rec false \\\n","        --cycle_w 10 --cmap_kind number"]},{"cell_type":"code","source":["!grep -R --line-number \"dex_imdb_wiki.caffemodel.pt\" training | head -n 20"],"metadata":{"id":"PWcWo5fZKIDE","executionInfo":{"status":"ok","timestamp":1746513557118,"user_tz":-180,"elapsed":107,"user":{"displayName":"Kutay Eroğlu","userId":"12425207516295289282"}}},"execution_count":18,"outputs":[]},{"cell_type":"code","source":["!grep -R --line-number \"classifier_kwargs\" training | head -n 20"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"du7qecVcUfU5","executionInfo":{"status":"ok","timestamp":1746513567914,"user_tz":-180,"elapsed":116,"user":{"displayName":"Kutay Eroğlu","userId":"12425207516295289282"}},"outputId":"6f26c912-5650-47a3-eb7d-37e82a868528"},"execution_count":19,"outputs":[{"output_type":"stream","name":"stdout","text":["grep: training/__pycache__/networks.cpython-311.pyc: binary file matches\n","training/networks.py:1319:    def __init__(self, classifier_kwargs, map_type=\"gb\"):\n","training/networks.py:1321:        self.model = dnnlib.util.construct_class_by_name(**classifier_kwargs)\n","training/networks.py:1448:        classifier_kwargs={},\n","training/networks.py:1555:                    classifier_kwargs, map_type=skip_grad_blur\n","training/networks.py:1596:                    classifier_kwargs, map_type=\"cam\"\n","training/loss.py:525:                 classifier_kwargs, classification_loss_kwargs, random_class_kwargs,\n","training/loss.py:537:        self.classifier = dnnlib.util.construct_class_by_name(**classifier_kwargs).to(device)\n","grep: training/__pycache__/loss.cpython-311.pyc: binary file matches\n"]}]},{"cell_type":"code","source":[],"metadata":{"id":"CtYnH0fCUh9m"},"execution_count":null,"outputs":[]}],"metadata":{"accelerator":"GPU","colab":{"gpuType":"A100","machine_shape":"hm","provenance":[],"authorship_tag":"ABX9TyMLGtX1YrNZhEDFtqBonSEZ"},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}